src/scripts.js

```diff
@@ -1,12 +1,129 @@
-var assert = require('assert')
-var ops = require('./opcodes')
-var typeForce = require('typeforce')
+var bufferutils = require('./bufferutils')
+var typeforce = require('typeforce')
+var types = require('./types')
 
+var ECSignature = require('./ecsignature')
 var ecurve = require('ecurve')
 var curve = ecurve.getCurveByName('secp256k1')
 
-var ECSignature = require('./ecsignature')
-var Script = require('./script')
+var OPS = require('./opcodes')
+var REVERSE_OPS = []
+for (var op in OPS) {
+  var code = OPS[op]
+  REVERSE_OPS[code] = op
+}
+
+function toASM (chunks) {
+  if (types.Buffer(chunks)) {
+    chunks = decompile(chunks)
+  }
+
+  return chunks.map(function (chunk) {
+    // data chunk
+    if (Buffer.isBuffer(chunk)) {
+      return chunk.toString('hex')
+
+    // opcode
+    } else {
+      return REVERSE_OPS[chunk]
+    }
+  }).join(' ')
+}
+
+function fromASM (asm) {
+  typeforce(types.String, asm)
+
+  var strChunks = asm.split(' ')
+  var chunks = strChunks.map(function (strChunk) {
+    // opcode
+    if (strChunk in OPS) {
+      return OPS[strChunk]
+
+    // data chunk
+    } else {
+      return new Buffer(strChunk, 'hex')
+    }
+  })
+
+  return compile(chunks)
+}
+
+function compile (chunks) {
```

Comment:
A `script` in this sense should be synonymous with a `Buffer`.
In all cases a `Script` was a valid, a `Buffer` should now be used.

`compile` is the only exception to this,  it takes an `Array` of chunks to be serialized into a `Buffer`.

`chunks` are the byte codes in their uncompiled form.
`toASM` takes a `Buffer` and transpiles it back to its human readable equivalent (ASM).
`fromASM` takes a string of human readable opcodes (ASM) and compiles it to a `Buffer`.

**For internal usage and performance reasons**, it is also possible to interchangeably use a `chunks` `Array` to avoid constant decompilation/compilation.